### DISCUSSION TITLE
Should exactly once work with batch kafka consumers?

### DISCUSSION THREAD

**Discussion by cookie-bytes** [BEagle1984/silverback#209] [Q&A] [answered]
Just wondering how exactly once works with batch kafka consumers, does it commit the offset to the offset store DB table once the whole batch is completed?

**Comment by BEagle1984** [comment, marked as answer]
Yes, exactly, the offsets are committed against both Kafka and the database when the subscriber successfully returns after having processed the whole batch.

**Reply by cookie-bytes** [reply]
Thanks for confirming
